cmd/state-mcp/internal/toolregistry/tools.go

```diff
@@ -37,4 +41,86 @@ func HelloWorldTool() Tool {
 			), nil
 		},
 	}
-}
\ No newline at end of file
+}
+
+type BuildNodesResponse struct {
+	Commit struct {
+		Build struct {
+			Nodes []BuildNode `json:"nodes"`
+		} `json:"build"`
+	} `json:"commit"`
+}
+
+type BuildNode struct {
+	Typename            string `json:"__typename"`
+	Name                string `json:"name"`
+	Namespace           string `json:"namespace"`
+	Version             string `json:"version"`
+	DisplayName         string `json:"displayName"`
+	LogURL              string `json:"logURL"`
+	Status              string `json:"status"`
+	LastBuildFinishedAt string `json:"lastBuildFinishedAt"`
+}
+
+func ProjectErrorsTool() Tool {
+	return Tool{
+		Category: ToolCategoryDebug,
+		Tool: mcp.NewTool(
+			"projecterrors",
+			mcp.WithDescription("Project errors tool - shows errors for a project"),
+			mcp.WithString("namespace", mcp.Description("Project namespace in format 'owner/project'")),
+		),
+		Handler: func(ctx context.Context, p *primer.Values, mcpRequest mcp.CallToolRequest) (*mcp.CallToolResult, error) {
+			namespace, _ := mcpRequest.RequireString("namespace")
+
+			ns, err := project.ParseNamespace(namespace)
+			if err != nil {
+				return mcp.NewToolResultError("Invalid namespace format. Use 'owner/project' format."), nil
+			}
+
+			runner := graphql.New(p, api.ServiceBuildPlanner)
```

Comment:
You can use: `gqlclient.NewWithOpts()`, eg.

https://github.com/ActiveState/cli/blob/169471b44a507f115252ba0a1eceb100a163e94d/pkg/platform/model/buildplanner/buildplanner.go#L43